ui/components/MesherySettings.js

```diff
@@ -134,205 +134,143 @@ TabContainer.propTypes = { children : PropTypes.node.isRequired, };
 
 
 //TODO: Tabs are hardcoded everywhere
-class MesherySettings extends React.Component {
-  constructor(props) {
-    super(props);
-    const {
-      k8sconfig, meshAdapters, grafana, prometheus, router : { asPath }
-    } = props;
-
-    this._isMounted = false;
-
-    //TODO: Extract this logic and clean it
-    let tabVal = 0, subTabVal = 0;
-    const splittedPath = asPath.split('#');
-    if (splittedPath.length >= 2 && splittedPath[1]) {
-      const subTabPath = splittedPath[1].split('/');
-
-      switch (subTabPath[0]) {
+const MesherySettings = (props) => {
+  const {
+    k8sconfig,
+    meshAdapters,
+    grafana,
+    prometheus,
+    classes,
+  } = props;
+
+  const [modelsCount, setModelsCount] = useState(0);
+  const [componentsCount, setComponentsCount] = useState(0);
+  const [relationshipsCount, setRelationshipsCount] = useState(0);
+  const [scannedGrafana, setScannedGrafana] = useState([]);
+  const [scannedPrometheus, setScannedPrometheus] = useState([]);
+  const [tabVal, setTabVal] = useState(0);
+  const [subTabVal, setSubTabVal] = useState(0);
+  const [isMeshConfigured] = useState(k8sconfig.clusterConfigured);
+  const systemResetPromptRef = useRef(null);
+
+
+  useEffect(() => {
+    if (
+      JSON.stringify(props.k8sconfig) !== JSON.stringify(k8sconfig) ||
+       JSON.stringify(props.meshAdapters) !== JSON.stringify(meshAdapters)
+    ) {
+
+      k8sconfig (props.k8sconfig)
```

Comment:
What does this line and the next 4 lines do? Can you elaborate a bit?

Reply:
Actually here we are checking the props.k8sconfig state conditionally that if this condition  {(JSON.stringify(props.k8sconfig) !== JSON.stringify(k8sconfig)} is true then props.k8config has changed . and if it's changed then we are updating the k8sconfig local variable .